src/strategyHelpers/blueGreen/promote.ts

```diff
@@ -0,0 +1,83 @@
+import * as core from '@actions/core'
+
+import {Kubectl} from '../../types/kubectl'
+import {
+   BlueGreenDeployment,
+   deployWithLabel,
+   NONE_LABEL_VALUE
+} from './blueGreenHelper'
+
+import {validateIngresses} from './ingressBlueGreenHelper'
+import {validateServicesState} from './serviceBlueGreenHelper'
+import {validateTrafficSplitsState} from './smiBlueGreenHelper'
+
+export async function promoteBlueGreenIngress(
+   kubectl: Kubectl,
+   manifestObjects
+): Promise<BlueGreenDeployment> {
+   //checking if anything to promote
+   let {areValid, invalidIngresses} = await validateIngresses(
```

Comment:
though it's only used in this case, I still don't want to throw from inside the validateIngress function just in case some situation arises where we encounter an ingress out of the correct state but want to just ignore it instead, feel like separating the validation/exception logic keeps the code more modular/keeps responsibility of "validating ingresses" to the validateIngress function and asks it to do nothing more

Reply:
okay sounds good